### PR TITLE
Add redis subchart

### LINKED ISSUE #113: Add a Redis deployment option
Since Renovate can be configured to use Redis for caching, I think it would be helpful to provide an optional Redis deployment and service that would be able to maintain its cache across different instantiations of the main CronJob.

I think a configuration block similar to this would work

```yaml
redis:
  enabled: false
  annotations: {}
  labels: {}
  image:
    repository: redis
    tag: "6"
    pullPolicy: IfNotPresent
```

I'd be happy to make a PR to implement this.

### PR DESCRIPTION
**Pull request by ArmaanT**
This PR:
* adds the ability to use the Bitnami Redis chart as a subchart so that renovate can be configured to use Redis for caching.
* Modifies the CI configuration to be able to handle subcharts 
* Adds an additional set of values with redis enabled for `ct` to lint and run tests on

Closes #113

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
Do we need to commit the subchart? I think this can be removed from repo.

**Comment by viceice** [comment]
It needs a github workflow change to add the helm repo for bitnami charts

**Comment by ArmaanT** [comment]
> Do we need to commit the subchart? I think this can be removed from repo.

I think it needs to be committed. If I delete the directory and try to helm install, I get the following error:

```
Release "renovate" does not exist. Installing it now.
Error: found in Chart.yaml, but missing in charts/ directory: redis
```

**Comment by JamieMagee** [comment]
I was just about to post your edit 😅 We need to add

```sh
helm repo add bitnami https://charts.bitnami.com/bitnami
helm dependency build
```

**Review by viceice — COMMENTED** [review]
Can you please try @JamieMagee plan.

**Comment by ArmaanT** [comment]
Okay, this should be ready for review now. I've updated the original PR description with an explanation of my changes.

I haven't tested the release workflow changes. However, it uses very similar logic to kubeval, so I'm reasonably confident it will work.

File: README.md [hunk 1]
```diff
@@ -29,3 +29,13 @@ helm install --generate-name --set renovate.config='\{\"token\":\"...\"\}' renov
 ```
 
 **NOTE**: `renovate.config` must be a valid Renovate [self-hosted configuration](https://docs.renovatebot.com/self-hosted-configuration/)
+
+## Contributing
+
+When using this repo locally or contributing to this repo, you will need to build the dependencies used for each helm chart. You can run the following commands to do so:
+
+```bash
+cd charts/renovate
```
Comment (viceice) [inline README.md:38]: Are you sure this needs to be run in chart subdir instead of repo root?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - master
     paths:
-      - 'charts/**'
+      - "charts/**"
```
Comment (viceice) [inline .github/workflows/release.yaml:8]: single quotes please

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -26,9 +26,19 @@ jobs:
...
+              (cd ${dir}; helm dependency build)
+          done
+
       - name: Run chart-releaser
         uses: helm/chart-releaser-action@v1.2.0
         with:
           version: v1.1.1 #renovate: datasource=github-releases depName=helm/chart-releaser
         env:
-          CR_TOKEN: '${{ secrets.CR_TOKEN }}'
+          CR_TOKEN: "${{ secrets.CR_TOKEN }}"
```
Comment (viceice) [inline .github/workflows/release.yaml:44]: single quotes please

File: README.md (hunk 1 above)
Comment (ArmaanT) [reply README.md]: The build command does, but not the repo add. Moved the order of those to make it more clear

File: .github/kubeval.sh [hunk 4]
```diff
@@ -12,7 +12,11 @@ SCHEMA_LOCATION="https://raw.githubusercontent.com/instrumenta/kubernetes-json-s
 curl --silent --show-error --fail --location --output /tmp/kubeval.tar.gz https://github.com/instrumenta/kubeval/releases/download/"${KUBEVAL_VERSION}"/kubeval-linux-amd64.tar.gz
 tar -xf /tmp/kubeval.tar.gz kubeval
 
+# add helm repos
+helm repo add bitname https://charts.bitnami.com/bitnami
```
[the remark this answers is not in the document]
Comment (ArmaanT) [reply .github/kubeval.sh]: nice catch

File: .github/workflows/release.yaml (hunk 2 above)
Comment (ArmaanT) [reply .github/workflows/release.yaml]: Fixed

File: .github/workflows/release.yaml (hunk 3 above)
Comment (ArmaanT) [reply .github/workflows/release.yaml]: Fixed

**Review by viceice — APPROVED** [review]
@JamieMagee what do you think?

File: charts/renovate/values.yaml [hunk 5]
```diff
@@ -71,3 +71,21 @@ resources:
   #  memory: 128Mi
 
 envFrom: []
+
+redis:
+  # Configuration for a Redis subchart. Additional documenation at
+  # https://github.com/bitnami/charts/tree/master/bitnami/redis.
```
Comment (JamieMagee) [inline charts/renovate/values.yaml:77]: This is a super small nit, but the full stop here breaks this link for me

```suggestion
  # https://github.com/bitnami/charts/tree/master/bitnami/redis
```

**Comment by JamieMagee** [comment]
Overall looks good. I just have 2 questions:
1. Is there any other configuration that a user might want to pass through to redis?
2. Is there any way to add the bitnami repo automatically? It's defined in `charts/renovate/Chart.yaml` and I would hope that `helm` could read that

**Comment by ArmaanT** [comment]
1. Users will be able to configure redis using any of the values supported by the bitnami chart. I just added a few defaults that made sense to me. By using the `redis.fullname` template the chart should be able to handle any custom configuration passed to redis.
2. I don't think so unfortunately. I removed the bitnami repo locally and tried to run `helm dependency build` but it complains that there's no repository definition for https://charts.bitnami.com/bitnami. Seems like helm only uses the definition to make sure the user has a repo with that URL and can't directly install from it

**Comment by JamieMagee** [comment]
Thanks for the answers. I had a look into 2 myself, and it appears that repositories are not added automatically due to security concerns.

LGTM

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**